### ISSUE TITLE
Original images no longer work

### ISSUE THREAD

**Issue by a84r7a3rga76fg** [ccloli/E-Hentai-Downloader#256]
New site update broke the script

**Comment by ccloli** [comment]
I've checked a gallery just now, and I've already disabled Source Nexus, seems it's not affected.

I also checked the source code when I noticed the update yesterday, and I'm sure the new update won't break the script, since the script detects the download link by the link target url instead of the link text.

Since the new update only affects the calculation of estimated cost, so I don't make an update for now. If you're having an issue for now, please check if you can download the original images directly, or post the log from the console by pressing F12 -> Console when the script is running.

![image](https://github.com/ccloli/E-Hentai-Downloader/assets/8115912/ea6438b0-70ae-40db-8dc1-3a68ab5f8208)

**Comment by ccloli** [comment]
> - H@H High-Capacity Ranges are now live. This change lets us efficiently take advantage of surplus H@H capacity for original images with some limitations.
> 
> Specifically, the system will now use H@H for original images smaller than 10 MiB in galleries that were posted in the last year, assuming that you haven't changed the H@H setting under "Image Load Settings" to "Default port clients only" or "No". Otherwise, it will still use the regular image servers.
> 
> This change applies for regular browsing when "Prefer original images" is selected and the file would normally be resampled at 2400x, when using the "Download Original" button, and when using the H@H Downloader.

A possible reason coule be the original image may delivered from H@H, but your network cannot connect to the node. However if that's the case, there's no way to fix it since the original URL doesn't have something like "reload if image fails" link on page viewing, which will force loaded from regular servers.

A possible workaround is to set "Image Load Settings" to "Default port clients only" as the update log said, but this may costs more (though from my test the cost is the same as normal, not sure if it's a bug or I made it wrong, that's another reason why I don't make an update).

> - When the "Default port clients only" setting is used, the cost for using the "Download original" button is increased.

**Comment by a84r7a3rga76fg** [comment]
Yes, it's H@H failing to serve the image properly. It sucks that the fallback URL is gone.  The only way to get H@H to serve failed images is change the IP address which is really inconvenient. I tried those workarounds but to no success.

**Comment by ccloli** [comment]
The problem is not limited to the script, but also for common users, since there's no way to reload a url that can be loaded from source server. Not sure if append a 'nl' param would fix the issue. Consider how the original image button works, I don't see there's a convenient way for EH itself to address this case.

But I remembered the script may cache the original download url, because the url won't change, but I'm not sure if I cached the redirect url, too. If that's the case, then the script will always retry that failed image link. Would review it later today.

**Comment by a84r7a3rga76fg** [comment]
jihcohulqbwpetewyuia.hath.network is the regular image server? I remember it as a regular exhentai URL with "nl" somewhere after it.

**Comment by ccloli** [comment]
> jihcohulqbwpetewyuia.hath.network is the regular image server? I remember it as a regular exhentai URL with "nl" somewhere after it.

Yep, the regular image server serves both original images and resample images. You can see a[screenshot](https://user-images.githubusercontent.com/8115912/114270065-e247be80-9a3c-11eb-8af4-5775b64e1070.png) I uploaded before (ignore the remark and highlights), in old behavior, when requesting the original image link, it'll redirect to a link that the url path is started with `/om/`

**Comment by a84r7a3rga76fg** [comment]
Can you toggle "Default port clients only" if "Any client" fails to load the original image and vice versa from the browser script?

**Comment by ccloli** [comment]
> Can you toggle "Default port clients only" if "Any client" fails to load the original image and vice versa from the browser script?

That would be too heavy for the script itself, and will affect all the other requests. Besides, the setting is buggy (towards to what it said in update log, but I think it's reasonable, since for this case the H@H is hosted at default port as it says in the setting), you can't say it'll return a regular server url at 100%. So I'd say no.

**Comment by ccloli** [comment]
v1.35 is released, luckilly you can append `nl` to original image url to let it loaded from regular server (even works for the new url format).

![image](https://github.com/ccloli/E-Hentai-Downloader/assets/8115912/3d4e924e-1e10-4aa0-a409-541beb0e4a11)

**Comment by Doofy420** [comment]
I've had some weird things going on as well, lots of images failing to load, and the downloader suddenly started including resized images in the zip files even though I had 'force download resized image'  unchecked. Not really sure how to replicate this.

**Comment by ccloli** [comment]
@Doofy420 See #259 instead, this issue is for tracking original image url may returns H@H url only.

BTW please update to 1.35 to see if it helps.

**Comment by ccloli** [comment]
@Doofy420 See #259 instead, this issue is for tracking original image url may returns H@H url only.

BTW please update to 1.35 to see if it helps.

**Comment by a84r7a3rga76fg** [comment]
Where do the numbers 37972-471877 after ?nl= come from in that screenshot?

**Comment by ccloli** [comment]
> Where do the numbers 37972-471877 after ?nl= come from in that screenshot?

Reload broken image
